pkg/apis/application/v1alpha1/types.go

```diff
@@ -72,6 +72,8 @@ type ApplicationSpec struct {
 	// Increasing will increase the space used to store the history, so we do not recommend increasing it.
 	// Default is 10.
 	RevisionHistoryLimit *int64 `json:"revisionHistoryLimit,omitempty" protobuf:"bytes,7,name=revisionHistoryLimit"`
+	// CreateNamespace indicates if namespace should be created for app resources
```

Comment:
I think this setting should part of `syncPolicy`. I'm proposing to reuse exiting `syncOptions`:

```
  # Sync policy
  syncPolicy:
    syncOptions:
    - CreateNamespace=true
```